Assignment refactoring (#22)
DareData/lp-foundations: assignments/assignment_4/README.md

Author: Getting used to doing code reviews is always good practise, but given that we extended a bit the amount of content, maybe we can skip it for this assignment... what do you think? I personally like the option of doing another code review :smiling_imp:

Reviewer: I think they should. It's a small PR and the code review shouldn't take much time. This is a boring task if the student doesn't use shortcuts and, in that case, they may want to go through a shortcut. They could, for example, only list 5 regions and call it a day. A code review may help us catch these instances

Author: Perfect!

# Assignment 4: OOP and VSCode Shortcuts

> The focus of this assignment is about OOP. In particular, we want to make sure that we only provide valid values for our countries. This is the perfect scenario to make use of the Enum class. Note that this is a special case of classes since they are rarely instantiated. 

Reviewer: ```suggestion
The focus of this assignment is about OOP. In particular, we want to make sure that we only provide valid values for our countries. This is the perfect scenario to make use of the Enum class. 
```

**Suggestion**: Remove this so we don't confuse students. Enum classes can be instantiated.. and it's how we can use enums to validate things: For example using `Region("PT")` will yield the class variable `Region.PT` which is treated as an instance, but is it? It can be confusing...

But going beyond that, our goal is to create the most annoying assignment ever written. The twist is that there is an easy way out: VSCode shortcuts.

> Don't forget to create a branch for this assignment.

## 1- Enums and Shortcuts

Passing a country as a string is not very safe (imprecise types are a code smell). We can use an enum to make sure that we only pass valid countries.

> 1. Create an `enum.Enum` called `Country` with possible country values. The idea behind this step is to find the most appropriate VSCode shortcut/s that you learned about in the [text editor](../../07_text_editors/README.md) module. 

Reviewer: ```suggestion
1. Create an `enum.Enum` called `Region` with **ALL** possible region values. You will have to extract the values from the pandas data frame and then copy the result to your `Region` class. Use the most appropriate VSCode shortcut/s that you learned about in the [text editor](../../07_text_editors/README.md) module to make your job easier! 
```

(making it explicit that we want all values)

> 2. Modify the necessary functions and tests to accept a `Country` instead of a string. Don't forget the type hints.
> 3. Finally, add a class method to `Country` that returns a list of all the _actual_ countries (so, it removes values like EU28, EFTA, etc). Add a test for this method. Again, make use of as many shortcuts as possible, it will make your life much easier.

Reviewer: ```suggestion
2. Modify the necessary functions and tests to accept a `Region` instead of a string. Don't forget the type hints.
3. Finally, add a class method to `Region` that returns a list of all the _actual_ countries (so, it removes values like EU28, EFTA, etc). Add a test for this method. Again, make use of as many shortcuts as possible, it will make your life much easier.
```

Renamed "Country" to "Region" since "EU28", etc are not countries. It also makes the useness of  the `get_countries` method more explicit
